raiden/tests/integration/test_echo_node.py

```diff
@@ -17,40 +20,49 @@
 def test_event_transfer_received_success(
         token_addresses,
         raiden_chain,
-        network_wait,
 ):
-    app0, app1, app2, receiver_app = raiden_chain
+    sender_apps = raiden_chain[:-1]
+    target_app = raiden_chain[-1]
+
     token_address = token_addresses[0]
+    registry_address = target_app.raiden.default_registry.address
+    target_address = target_app.raiden.address
 
-    expected = dict()
+    message_handler = WaitForMessage()
+    target_app.raiden.message_handler = message_handler
 
-    for num, app in enumerate([app0, app1, app2]):
-        amount = 1 + num
-        payment_status = RaidenAPI(app.raiden).transfer_async(
-            app.raiden.default_registry.address,
-            token_address,
-            amount,
-            receiver_app.raiden.address,
-        )
-        payment_status.payment_done.wait(timeout=20)
-        expected[app.raiden.address] = amount
+    wait_for = list()
+    for amount, app in enumerate(sender_apps, 1):
+        secret = random_secret()
 
-    # sleep is for the receiver's node to have time to process all events
-    gevent.sleep(1)
+        wait = message_handler.wait_for_message(
+            Unlock,
+            {'secret': secret},
+        )
+        wait_for.append((wait, app.raiden.address, amount))
 
-    def test_events(amount, address):
-        return search_for_item(
-            receiver_app.raiden.wal.storage.get_events(),
-            EventPaymentReceivedSuccess,
-            {'amount': amount, 'initiator': address},
+        RaidenAPI(app.raiden).transfer_async(
+            registry_address=registry_address,
+            token_address=token_address,
+            amount=amount,
+            identifier=amount,
+            target=target_address,
+            secret=to_hex(secret),
+            secret_hash=to_hex(sha3(secret)),
         )
 
-    amounts = [1, 2, 3]
-    addrs = [app0.raiden.address, app1.raiden.address, app2.raiden.address]
-    for amount, address in zip(amounts, addrs):
-        assert wait_until(
-            lambda: test_events(amount, address),
-            network_wait,
+    for wait, sender, amount in wait_for:
+        wait.wait()
+        assert search_for_item(
+            target_app.raiden.wal.storage.get_events(),
+            EventPaymentReceivedSuccess,
+            {
+                'amount': amount,
+                'identifier': amount,
+                'initiator': sender,
+                'payment_network_identifier': registry_address,
+                # 'token_network_identifier': ,
```

Comment:
Nitpick: this can be removed.

Reply:
I left it there to have a full list of all the events attributes, but I can remove it.

What about the other test that is pretty much the same as this?

Reply:
I think it makes sense for us to remove duplicate tests. 

It's your call.